client-js/app/elements/expandable-input.html

```diff
@@ -0,0 +1,97 @@
+<!-- 
+Expandable input. Takes a registry key value and shortens it to fit on the 
+screen. When clicked on reveals an editable text box which is shifted into 
+focus. Enter fires 'iron-form-submit' which sends the key, value pair as 
+event.detail
+ -->
+
+<dom-module id="expandable-input">
+  <style>
+  #long{
+    visibility: hidden;
+    position: absolute;
+    width: 80%;
+  }
+  </style>
+  <template >
+    <iron-a11y-keys keys="enter" on-keys-pressed="submitForm">
+    <!-- TODO remap newline from enter to "ctrl enter" -->
+      <div id="long">
+        <textarea id="input" value="{{value::input}}" rows="[[compRows]]" cols="[[cols]]"></textarea>
```

Comment:
This binds the value in the field to the input of the area, this may not be necessary anymore as it is left over from when I was using an ```iron-input``` I can try without it.
https://github.com/PolymerElements/iron-input